### PR TITLE
Issue 18896 Replace deprecated Button component with new Button component

### LINKED ISSUE #18718: Deprecate old `UrlIcon` in favour of appropriate `Avatar`
### Description
To prevent further tech debt and inconsistencies the ui `UrlIcon` component can be deprecated in favour of one of the avatar components `AvatarNetwork`, `AvatarToken` or `AvatarFavicon`

`UrlIcon`(old): https://metamask.github.io/metamask-storybook/?path=/story/components-ui-urlicon--bat
`AvatarNetwork`(new): https://metamask.github.io/metamask-storybook/?path=/story/components-componentlibrary-avatarnetwork--default-story
`AvatarToken`(new): https://metamask.github.io/metamask-storybook/?path=/story/components-componentlibrary-avatartoken--default-story
`AvatarToken`(new): https://metamask.github.io/metamask-storybook/?path=/story/components-componentlibrary-avatarfavicon--default-story

### Technical Details
- Create good first issue to replace all old components with new [LINK TO TEMPLATE TBC]()
- Add deprecation notice to storybook
- Add deprecation message above `SiteIcon` component code

### Acceptance Criteria
- Old component has been deprecated, deprecation indicator shows up in VS Code
- Good first issue has been created for the community to contribute to reducing tech debt and improving the UI consistency of MetaMask

### LINKED ISSUE #18896: Replace deprecated Button component with new Button component
### Description

Currently, the extension is using an outdated `Button` component, which needs to be replaced with the new `Button` component. The new Button component is available from `ui/components/component-library`.

This is a massive undertaking by itself and creating a single PR would be too large. Smaller PRs can be submitted against this issue to ensure easier review and gradual improvements. **PRs should contain no more than 3 files**

### Technical Details

- The old Button component (`ui/components/ui/button/button.component.js`)
- The new Button component (`ui/components/component-library/button/button.js`) [storybook documentation](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-button--default-story#button)
- Component APIs are slightly different so ensure all props have been migrated appropriately

### Acceptance Criteria

- The old Button component (`ui/components/ui/button/button.component.js`) is completely replaced with the new Button component (`ui/components/component-library/button/button.js`) throughout all files included in the PR.
- The component APIs are updated to reflect the changes in the new Button component.
- Each Pull Request (PR) should include no more than 3 files with Button replacements.
- The code changes should pass Jest tests, linting, and all CI checks without any errors.
- The PR must include before and after screenshots of the UI to ensure there are no visual regressions.

If the acceptance criteria are not met, PRs may be closed.

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with running the extension locally, have knowledge of React, component props, Jest tests, linting, and Storybook, and want to contribute to improving the cohesiveness of UI in the extension

### PR DESCRIPTION
**Pull request by ujwalkumar1995**
## Explanation
As part of this PR, I have replaced the depracated button component (ui/components/ui/button/button.component.js) with ui/components/component-library/button/button.js
* Fixes #18896 
* Fixes #18718

### Before

Screenshot 2023-06-13 at 5 08 21 PM

### After

Screenshot 2023-06-13 at 5 08 43 PM

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by ujwalkumar1995** [comment]
Hey @georgewrmarshall 

I have submitted this PR for fixing one of the files for issue #18896 
I can see in the screenshot of before and after that there is some styling difference between the old and new buttons. I am not sure if this is intended as per some new design as I have not changed any styling as part of my changes. 
Please let me know if this needs to be fixed and I will try to explore further.

Thanks in advance!

File: ui/components/component-library/index.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ export {
 export { AvatarFavicon, AVATAR_FAVICON_SIZES } from './avatar-favicon';
 export { AvatarIcon, AVATAR_ICON_SIZES } from './avatar-icon';
 export { AvatarNetwork, AVATAR_NETWORK_SIZES } from './avatar-network';
-export { AvatarToken } from './avatar-token';
+export { AvatarToken, AVATAR_TOKEN_SIZES } from './avatar-token';
```
Comment (georgewrmarshall) [inline ui/components/component-library/index.js:10]: Wasn't exported from root

File: ui/components/component-library/modal-content/modal-content.tsx [hunk 2]
```diff
@@ -90,9 +88,9 @@ export const ModalContent = forwardRef(
             aria-modal="true"
             backgroundColor={BackgroundColor.backgroundDefault}
             borderRadius={BorderRadius.LG}
-            width={BLOCK_SIZES.FULL}
-            marginTop={12}
-            marginBottom={12}
+            width={BlockSize.Full}
+            marginTop={[null, 8, 12]}
+            marginBottom={[null, 8, 12]}
```
Comment (georgewrmarshall) [inline ui/components/component-library/modal-content/modal-content.tsx:93]: Improves margin using array props


https://github.com/MetaMask/metamask-extension/assets/8112138/75a5119d-4cc1-4c8a-be6f-07b5ed5cd366

File: ui/components/component-library/modal-content/modal-content.types.ts [hunk 3]
```diff
@@ -29,5 +29,5 @@ export interface ModalContentProps extends BoxProps {
   /**
    * Additional props to pass to the dialog node inside of ModalContent component
    */
-  modalDialogProps?: BoxProps | React.HTMLAttributes<HTMLElement>;
+  modalDialogProps?: any;
 }
```
Comment (georgewrmarshall) [inline ui/components/component-library/modal-content/modal-content.types.ts:33]: Temporary fix for ModalContent to use new Box component

File: ui/components/ui/url-icon/url-icon.js [hunk 4]
```diff
@@ -3,6 +3,15 @@ import PropTypes from 'prop-types';
...
 
+/**
+ * @deprecated `<UrlIcon />` has been deprecated in favor of one of the avatar components `<AvatarNetwork />`, `<AvatarToken/>` or `<AvatarFavicon/>`
+ * component in ./ui/components/component-library/avatar-token/
+ * See storybook documentation for BannerAlert here:
+ * {@see {@link https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-avatartoken--docs}}
+ *
+ * Help to replace `UrlIcon` with `AvatarToken` by submitting a PR
+ */
+
```
Comment (georgewrmarshall) [inline ui/components/ui/url-icon/url-icon.js:14]: Adding deprecation message to old UrlIcon component

Screenshot 2023-06-13 at 5 13 15 PM

File: ui/pages/swaps/import-token/index.scss
Comment (georgewrmarshall) [inline ui/pages/swaps/import-token/index.scss:1]: Removing unneeded CSS

File: ui/pages/swaps/index.scss [hunk 5]
```diff
@@ -14,7 +14,6 @@
 @import 'slippage-buttons/index';
 @import 'swaps-footer/index';
 @import 'view-quote/index';
-@import 'import-token/index';
 @import 'create-new-swap/index';
```
Comment (georgewrmarshall) [inline ui/pages/swaps/index.scss:17]: Removing unused

File: ui/pages/swaps/import-token/import-token.js [hunk 6]
```diff
@@ -1,89 +1,102 @@
...
+            size={BUTTON_SIZES.LG}
+            onClick={onImportTokenClick}
+            data-testid="page-container__import-button"
+          >
+            {t('import')}
+          </Button>
+        </Box>
+      </ModalContent>
+    </Modal>
   );
```
Comment (georgewrmarshall) [inline ui/pages/swaps/import-token/import-token.js:100]: Updating to use 100% component-library components

**Review by georgewrmarshall — DISMISSED** [review]

**Comment by georgewrmarshall** [comment]
Hey @ujwalkumar1995, would you be able to update this PR I believe it is close so it would be great if we can get it merged

**Comment by ujwalkumar1995** [comment]
Hey @georgewrmarshall 
Sorry I completely forgot about this PR. I could not see any open review commits but I could see a merge conflict. I have resolved it. Please review.

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution @ujwalkumar1995

**Review by garrettbear — DISMISSED** [review]

**Review by garrettbear — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**